account_payment_order/models/account_payment_order.py

```diff
@@ -451,3 +451,14 @@ def action_move_journal_line(self):
         ctx.update({"search_default_misc_filter": 0})
         action["context"] = ctx
         return action
+
+    def action_view_payments(self):
+        self.ensure_one()
+        return {
+            "name": _("Payments"),
+            "type": "ir.actions.act_window",
+            "res_model": "account.payment",
+            "view_mode": "list,form",
+            "domain": [("payment_order_id", "=", self.id)],
+            "context": {"default_payment_order_id": self.id},
+        }
```

Comment:
There could be a better way to do this, without losing the views, since payments can be displayed in `kanban, graph, and activities`, not only in list and form views.

<img width="172" height="58" alt="image" src="https://github.com/user-attachments/assets/3b495ff8-6462-4fc4-8e8d-3071c9f10fa2" />

```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id(
            "account.action_account_payments"
        )
        action.update(
            "domain": [("payment_order_id", "=", self.id)],
            "context": {"default_payment_order_id": self.id},
        })
        return action
```